security/cmd/istio_ca/main.go

```diff
@@ -328,24 +339,41 @@ func runCA() {
 		}
 	}
 
-	log.Info("Istio CA has started")
+	monitorErrCh := make(chan error)
+	// Start the monitoring server.
+	if opts.monitoringPort > 0 {
+		monitor, mErr := monitoring.NewMonitor(opts.monitoringPort, opts.enableProfiling)
+		if mErr != nil {
+			fatalf("Unable to setup monitoring: %v", mErr)
+		}
+		go monitor.Start(monitorErrCh)
+		log.Info("Citadel monitor has started.")
+		defer monitor.Close()
```

Comment:
is `runCA()` expected to not return until server shutdown?

Reply:
Yes. Any concerns?